src/operator/convolution.cu

```diff
@@ -28,15 +28,63 @@ Operator* CreateOp<gpu>(ConvolutionParam param, int dtype,
     return op;
   }
 #if MXNET_USE_CUDNN == 1
-  if (param.dilate.Size() == 1 && !param.cudnn_off) {
-    MSHADOW_REAL_TYPE_SWITCH(dtype, DType, {
-      op = new CuDNNConvolutionOp<DType>(param, *in_shape, *out_shape, ctx);
-    })
-  } else {
-    MSHADOW_REAL_TYPE_SWITCH(dtype, DType, {
+  // The NVIDIA Pascal architecture was the first to include 16-bit ALUs.
+  // Thus, when the framework is compiled with MSHADOW_USE_PASCAL == 1, we
+  // perform the convolution calculation in 16-bit when the tensor type is
+  // also 16-bit.  For NVIDIA architectures earlier than Pascal (so Maxwell
+  // and Kepler), the computation precision is always at least 32-bits.
+#if MSHADOW_USE_PASCAL == 1
+  // true fp16
+  int desired_forward_compute_type = dtype;
```

Comment:
@DickJC123 can we detect this in runtime so we don't need to provide separate binaries for pascal/maxwell gpu?